elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchTable.java

```diff
@@ -128,8 +346,74 @@ private ElasticsearchSearchResult httpRequest(String query) throws IOException {
     }
 
     try (InputStream is = response.getEntity().getContent()) {
-      return mapper.readValue(is, ElasticsearchSearchResult.class);
+      return (ObjectNode) mapper.readTree(is);
```

Comment:
I think it would be better to cache an `ObjectReader` and use that to consume the input stream directly to avoid unnecessary copying.